server/channels/store/sqlstore/post_store.go

```diff
@@ -2896,6 +2896,35 @@ func (s *SqlPostStore) deleteThread(transaction *sqlxTxWrapper, postId string, d
 		return errors.Wrapf(err, "failed to mark thread for root post %s as deleted", postId)
 	}
 
+	return s.deleteThreadFiles(transaction, postId, deleteAtTime)
+}
+
+func (s *SqlPostStore) deleteThreadFiles(transaction *sqlxTxWrapper, postID string, deleteAtTime int64) error {
+	var err error
+	if s.DriverName() == model.DatabaseDriverPostgres {
+		queryString, args, err1 := s.getQueryBuilder().
+			Update("FileInfo").
+			Set("DeleteAt", deleteAtTime).
+			From("Posts").
+			Where(sq.Expr("FileInfo.PostId = Posts.Id AND Posts.RootId = ?", postID)).
+			ToSql()
+		if err1 != nil {
+			return errors.Wrapf(err1, "failed to create SQL query to mark files of thread post %s as deleted", postID)
+		}
+
+		_, err = transaction.Exec(queryString, args...)
+	} else {
+		_, err = transaction.NamedExec(`
```

Comment:
Why `NamedExec` here and query builder above? Seems like we could simplify the error handling (i.e. `err1` juggling) if we just did it consistently?

Reply:
Unfortunately, [Squirrel does not support using "join" together with "update"](https://github.com/Masterminds/squirrel/issues/257). Also, MySQL and PostgreSQL do not work in the same way in this case. We could consider using a query with a subquery to collect the IDs that will be compatible with both, but I think this adds complexity as well and possibly a performance hit. Would you happen to have something better in mind?

Reply:
Oh, right, I'd forgotten about some of the upstream limitations. However, thanks to our fork of squirrel, I think we can do this with multiple tables in the `FROM` statement, e.g.

https://github.com/mattermost/mattermost-server/blob/609ab9c765be14708cfb1d25520cbad7934c325b/server/channels/store/sqlstore/thread_store.go#L634-L652

Reply:
Oh, I didn't realize we had that change in our fork. Thanks for the heads-up!